### PR TITLE
vaults: init at 0.6.1

### PR DESCRIPTION
**Pull request by benneti**
## Description of changes

A package that fits in with the gnome design to create and mount encrypted "stores" using encfs or gocryptfs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/va/vaults/package.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+stdenv.mkDerivation rec {
+  pname = "vaults";
+  version = "0.6.1";
+
+  src = fetchFromGitHub {
+    owner = "mpobaschnig";
+    repo = "Vaults";
+    rev = version;
+    fetchSubmodules = false;
+    sha256 = "sha256-x7NoYQ+ol/j8LMazL4A0jDi/I4ajRNUzOpShLB0eHUg=";
```
Comment (Lurkki14) [inline pkgs/by-name/va/vaults/package.nix:32]:
```suggestion
    hash = "sha256-x7NoYQ+ol/j8LMazL4A0jDi/I4ajRNUzOpShLB0eHUg=";
```

File: pkgs/by-name/va/vaults/package.nix [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    wayland
+    gocryptfs
+    cryfs
+  ];
+
+  meta = with lib; {
+    description = "GTK frontend for encrypted vaults supporting gocryptfs and CryFS for encryption.";
+    homepage = "https://mpobaschnig.github.io/vaults/";
+    changelog = "https://github.com/mpobaschnig/vaults/releases/tag/${version}";
+    license = licenses.gpl3;
```
Comment (Lurkki14) [inline pkgs/by-name/va/vaults/package.nix:82]:
```suggestion
    license = licenses.gpl3Plus;
```
Just `licenses.gpl3` is deprecated

File: pkgs/by-name/va/vaults/package.nix [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    maintainers = with maintainers; [ benneti ];
```
Comment (Lurkki14) [inline pkgs/by-name/va/vaults/package.nix:83]: `meta.mainProgram` should be set

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/va/vaults/package.nix (hunk 3 above)
Comment (benneti) [reply pkgs/by-name/va/vaults/package.nix:83]: done, I did not expect it should be set if it is the same as pname (as it defaults to pname afaik)

**Comment by benneti** [comment]
thank you for the review, implemented all of the suggestions and squashed into a single commit.

File: pkgs/by-name/va/vaults/package.nix [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    meson
+    ninja
+    cmake
+    pkg-config
+    python3
+    wrapGAppsHook
+    cargo
+    rustc
+    rustPlatform.cargoSetupHook
+  ];
```
Comment (Lurkki14) [inline pkgs/by-name/va/vaults/package.nix:68]:
```suggestion
  nativeBuildInputs = [
    desktop-file-utils
    meson
    ninja
    cmake
    pkg-config
    wrapGAppsHook
    cargo
    rustc
    rustPlatform.cargoSetupHook
  ];
```
The removed items only need to be in `buildInputs`

File: pkgs/by-name/va/vaults/package.nix [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    repo = "Vaults";
+    rev = version;
+    fetchSubmodules = false;
+    hash = "sha256-x7NoYQ+ol/j8LMazL4A0jDi/I4ajRNUzOpShLB0eHUg=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    sha256 = "sha256-fDUsbeJzgqjvNCNce1FnvvqZfXu0X5Knpan4Q+PYe+Q=";
```
Comment (Lurkki14) [inline pkgs/by-name/va/vaults/package.nix:38]:
```suggestion
    hash = "sha256-fDUsbeJzgqjvNCNce1FnvvqZfXu0X5Knpan4Q+PYe+Q=";
```

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

**Review by Lurkki14 — APPROVED** [review]

File: pkgs/by-name/va/vaults/package.nix [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "vaults";
+  version = "0.6.1";
+
+  src = fetchFromGitHub {
+    owner = "mpobaschnig";
+    repo = "Vaults";
+    rev = version;
+    fetchSubmodules = false;
```
Comment (AndersonTorres) [inline pkgs/by-name/va/vaults/package.nix:31]:
```suggestion
```

File: pkgs/by-name/va/vaults/package.nix [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    python3
+    glib
+    libadwaita
+    wayland
+    gocryptfs
+    cryfs
+  ];
+
+  meta = with lib; {
+    description = "GTK frontend for encrypted vaults supporting gocryptfs and CryFS for encryption.";
```
Comment (AndersonTorres) [inline pkgs/by-name/va/vaults/package.nix:79]:
```suggestion
    description = "GTK frontend for encrypted vaults supporting gocryptfs and CryFS for encryption";
```

File: pkgs/by-name/va/vaults/package.nix [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    homepage = "https://mpobaschnig.github.io/vaults/";
+    changelog = "https://github.com/mpobaschnig/vaults/releases/tag/${version}";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ benneti ];
+    meta.mainProgram = pname;
```
Comment (AndersonTorres) [inline pkgs/by-name/va/vaults/package.nix:84]:
```suggestion
    mainProgram = "vaults";
```

File: pkgs/by-name/va/vaults/package.nix [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix PATH : "${lib.makeBinPath [ gocryptfs cryfs ]}"
+    )
+  '';
+
+  nativeBuildInputs = [
+    desktop-file-utils
+    meson
```
Comment (AndersonTorres) [inline pkgs/by-name/va/vaults/package.nix:57]: why are you using meson, cmake and ninja?

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/va/vaults/package.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/by-name/va/vaults/package.nix]: The "defaults to pname" is now urban legend.

File: pkgs/by-name/va/vaults/package.nix (hunk 8 above)
Comment (benneti) [reply pkgs/by-name/va/vaults/package.nix:84]: thanks is `mainProgram = pname` also fine?

File: pkgs/by-name/va/vaults/package.nix (hunk 9 above)
Comment (benneti) [reply pkgs/by-name/va/vaults/package.nix:57]: meson and ninja both are necessary for me to build cmake I might have forgotten to remove when copying some derivation to start from...

File: pkgs/by-name/va/vaults/package.nix (hunk 8 above)
Comment (AndersonTorres) [reply pkgs/by-name/va/vaults/package.nix]: Parameterizing `pname` is not a good idea. The explanation is a bit convoluted, but, summarizing, one of the reasons is that argument overriding is harder in presence of too much attributes being bound like this.

**Comment by benneti** [comment]
set platform to linux as ofborg fails on macos (and squash commits).

**Comment by benneti** [comment]
closing in favor of https://github.com/NixOS/nixpkgs/pull/287260

**Pull request closed without merging**